### PR TITLE
Add hts precompile approval/allowance tests

### LINKED ISSUE #399: Add htsPrecompile allowance/approval acceptance test
### Problem

No allowance/approval HTS precompile support exists

### Solution

Add acceptance test support for the HIP 514 methods below. Basic initial coverage is sufficient

- `allowance(address,address,address)`
- `getApproved(address,uint256)`
- `isApprovedForAll(address,address,address)`
- `setApprovalForAll(address,address,bool)`
- `approve(address,address,uint256)`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
Signed-off-by: nikolay <n.atanasow94@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #399

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/server/tests/acceptance/htsPrecompile.spec.ts [hunk 1]
```diff
@@ -34,60 +34,147 @@ describe('HTS Precompile Acceptance Tests', async function() {
...
+
     expect(balanceBefore + amount).to.equal(balanceAfter);
   });
+
+  it('should be able to check allowance', async function() {
+    const baseHTSContract = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const tx = await baseHTSContract.allowancePublic(HTSTokenContractAddress, baseHTSContractAddress, accounts[2].wallet.address);
+    const { responseCode } = (await tx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args;
+
+    expect(responseCode).to.equal(TX_SUCCESS_CODE);
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/htsPrecompile.spec.ts:136]: Move this check into the next test `should be able to approve anyone to spend tokens` and you should be able to remove this test.
For unit tests it's good to have it broken out but for E2E acceptance you can't do anything without success and it calls cost HBAR so no need for the breakout

File: packages/server/tests/acceptance/htsPrecompile.spec.ts [hunk 2]
```diff
@@ -34,60 +34,147 @@ describe('HTS Precompile Acceptance Tests', async function() {
...
+    expect(beforeAmount).to.equal(0);
+    expect(afterAmount).to.equal(amount);
+  });
+
+  it('should be able to execute setApprovalForAll on nft', async function() {
+    const baseHTSContract = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const tx = (await baseHTSContract.isApprovedForAllPublic(NftHTSTokenContractAddress, baseHTSContractAddress, accounts[1].wallet.address));
+    const { responseCode } = (await tx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args;
+
+    expect(responseCode).to.equal(TX_SUCCESS_CODE);
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/htsPrecompile.spec.ts:160]: Move this check into the next test `should be able to execute setApprovalForAllPublic` and you should be able to remove this test.
For unit tests it's good to have it broken out but for E2E acceptance you can't do anything without success and it calls cost HBAR so no need for the breakout

File: packages/server/tests/contracts/BaseHTS.sol [hunk 3]
```diff
@@ -67,4 +114,38 @@ contract BaseHTS is FeeHelper {
             revert();
         }
     }
+
+    function getApprovedPublic(address token, uint256 serialNumber) public returns (int responseCode, address approved)
```
Comment (Nana-EC) [inline packages/server/tests/contracts/BaseHTS.sol:118]: NO test coverage for this method

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good.
Big bump on cost and missing test coverage case

File: packages/server/tests/acceptance/htsPrecompile.spec.ts [hunk 4]
```diff
@@ -34,60 +34,161 @@ describe('HTS Precompile Acceptance Tests', async function() {
...
+    return tokenAddress;
   }
 
   it('should associate to a token', async function() {
-    const baseHTSContractReceiverWalletFirst = new ethers.Contract(baseHTSContract.address, BaseHTSJson.abi, accounts[1].wallet);
-    expect(baseHTSContractReceiverWalletFirst.associateTokenTo(accounts[1].wallet.address, HTSTokenContract.address, { gasLimit: 10000000 }))
-      .to.not.be.reverted;
+    const baseHTSContractOwner = new ethers.Contract(BaseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const txCO = await baseHTSContractOwner.associateTokenPublic(BaseHTSContractAddress, HTSTokenContractAddress, { gasLimit: 10000000 });
+    expect((await txCO.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.equal(22);
```
Comment (ar-conmit) [inline packages/server/tests/acceptance/htsPrecompile.spec.ts:88]: nit: status code `22` can be replaced with the const var `TX_SUCCESS_CODE`

**Review by Nana-EC — APPROVED** [review]
LG

File: packages/server/tests/acceptance/htsPrecompile.spec.ts (hunk 4 above)
Comment (Nana-EC) [reply packages/server/tests/acceptance/htsPrecompile.spec.ts:88]: About to merge this to unblock you so you can make this modification in your PR on top of this

**Pull request merged**